world.js

```diff
@@ -32,6 +32,63 @@ const localMatrix = new THREE.Matrix4();
 const localMatrix2 = new THREE.Matrix4();
 const localRaycaster = new THREE.Raycaster();
 
+let pendingMonetizationStart = [];
+setInterval(() => {
```

Comment:
Pending monetizations can just be queued synchronously, right? I don't quite understand this delay, unless it's intended for audio effects.

Reply:
There is a ~5ms delay between the script loading in that creates the custom `eventTarget` and the code running.. which is why this has to keep checking to see if the custom `eventTarget` exists yet.

39ca41e 

![11msasync](https://user-images.githubusercontent.com/29695350/101445903-ebd2e580-38e7-11eb-9f80-36e130e10d34.PNG)
![lolasync](https://user-images.githubusercontent.com/29695350/101445906-ed041280-38e7-11eb-9c99-3fc590e2e1fc.PNG)

Reply:
It sounds like this code should go wherever the conditions for its triggering change, not on an interval. If that is after the app script is loaded in the runtime then maybe this update should happen there?

https://github.com/webaverse/app/blob/111d3b7d762425d5c7b605a0b3bb7dfb9701a261/runtime.js#L473

Reply:
worked perfectly 
0f6f10d